clientv3/client.go

```diff
@@ -507,10 +507,16 @@ func toErr(ctx context.Context, err error) error {
 	code := ev.Code()
 	switch code {
 	case codes.DeadlineExceeded:
-		fallthrough
+		if ctx.Err() != nil {
+			err = ctx.Err()
+		} else if strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
why do we need this line?

Reply:
Actually believe we don't need this.

1. context from client times out while selecting on `linearizableReadNotify`
2. return `context.DeadlineExceeded` (as of this patch)
3. gRPC returns `codes.DeadlineExceeded` error of `grpc/*status.statusError` to the client

So it will be `ctx.Err() != nil` anyway.